lib/readline.js

```diff
@@ -500,6 +501,28 @@ Interface.prototype._insertString = function(c) {
     // a hack to get the line refreshed if it's needed
     this._moveCursor(0);
   }
+  // emit current line for generating preview
+  this.emit('buffer', this.line);
+};
+
+Interface.prototype.appendPreview = function(result) {
+  this.previewResult = `\u001b[90m // ${result}\u001b[39m`;
```

Comment:
Colors might not be supported. Please check that first before using them.

Reply:
Ah, I wasn't aware of this color part. Will check them and update the PR.

Reply:
Have added logic below to handle the color part, PTAL..